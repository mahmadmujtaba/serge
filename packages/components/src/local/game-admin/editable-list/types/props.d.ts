export interface Item {
  name: string
  [property: string]: any
}

export default interface PropTypes {
  items: Array<Item>
  onClick?: (item: Item, e?: any) => void
  onChange?: (items: Array<Item>) => void
  onCreate?: () => void
  title?: string
}
