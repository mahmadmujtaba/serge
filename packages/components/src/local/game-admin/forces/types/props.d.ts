import { ForceData } from '@serge/custom-types'

export default interface PropTypes {
  forces: Array<ForceData>
  onChange: (forces: Array<ForceData>) => void
  onSave?: () => void
}

export { ForceData } from '@serge/custom-types'
export { Role } from '@serge/custom-types'
