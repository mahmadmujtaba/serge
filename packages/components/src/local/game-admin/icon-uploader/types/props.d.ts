export default interface PropTypes {
  onChange?: (src: string) => void
  background?: string
  icon?: string
  limit: number
}
