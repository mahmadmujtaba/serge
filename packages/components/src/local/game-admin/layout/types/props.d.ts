import { Wargame } from '@serge/custom-types';

export default interface PropTypes {
  /* half section title */
  type?: 'submit' | 'reset' | 'button'
  size?: 's' | 'm'
  onClick?: any
  disabled?: boolean
  wargame?: Wargame
  tabs?: Array<string>
}
