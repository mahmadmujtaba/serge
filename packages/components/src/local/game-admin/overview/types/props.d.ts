import { WargameOverview } from '@serge/custom-types'

export default interface PropTypes {
  overview: WargameOverview
  onChange: (nextOverview: WargameOverview) => void
  onSave?: () => void
}

export { WargameOverview } from '@serge/custom-types'
