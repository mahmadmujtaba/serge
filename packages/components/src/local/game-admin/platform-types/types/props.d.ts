import { PlatformType } from '@serge/custom-types'

export default interface PropTypes {
  platformType?: PlatformType
  onChange: (platformType: PlatformType) => void
  onSave?: () => void
}
