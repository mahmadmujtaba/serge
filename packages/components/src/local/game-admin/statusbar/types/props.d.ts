import { Wargame } from '@serge/custom-types';

export default interface PropTypes {
  /* half section title */
  href?: string
  onChange?: (e?: any) => void
  wargame: Wargame
}
