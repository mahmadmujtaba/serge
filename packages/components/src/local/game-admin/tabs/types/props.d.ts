export default interface PropTypes {
  /* half section title */
  onChange?: (selectedTab: string, e?: any) => void
  tabs: Array<string>
  activeTab?: string
}
