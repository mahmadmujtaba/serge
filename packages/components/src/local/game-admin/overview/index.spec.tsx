/* global it expect */

import React from 'react'
import renderer from 'react-test-renderer'

import Overview from './index'
import { WargameExportedMock } from '@serge/mocks'
import { WargameOverview } from './types/props'

const handleChange = (nextOverview: WargameOverview) => {
  console.log(nextOverview)
}

it('overview renders correctly', () => {
  const tree = renderer
    .create(<Overview overview={WargameExportedMock.data.overview} onChange={handleChange} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
