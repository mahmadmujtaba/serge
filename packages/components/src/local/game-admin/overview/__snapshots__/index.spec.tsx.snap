// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overview renders correctly 1`] = `
<div
  className="main"
>
  <div
    className="row"
  >
    <div
      className="col"
    />
    <div
      className="actions"
    >
      <button
        className="main"
        onClick={[Function]}
        type="button"
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col section"
    >
      <div
        className="main"
      >
        <div
          className="placeholder"
        >
          Game description & objectives
        </div>
        <div
          className="content"
        >
          <textarea
            className="textarea"
            onChange={[Function]}
            rows={8}
            value="Sample MWARC Wargame"
          />
        </div>
      </div>
    </div>
    <div
      className="col section"
    >
      <div
        className="main"
      >
        <div
          className="placeholder"
        />
        <div
          className="content"
        >
          <div
            className="group"
          >
            <label
              className="label label-first"
              htmlFor="gameTurnTime"
            >
              Wargame turn time (DD HH MM SS)
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline"
            >
              <input
                className="MuiInputBase-input MuiInput-input"
                id="gameTurnTime"
                mask="11 11 11 11"
                maxLength={11}
                name="gameTurnTime"
                onChange={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onPaste={[Function]}
                placeholder="DD HH MM SS"
                size={11}
                value="00 01 30 00"
              />
            </div>
          </div>
          <div
            className="group"
          >
            <label
              className="label"
              htmlFor="realtimeTurnTime"
            >
              Real time planning allowance (DD HH MM SS)
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline"
            >
              <input
                className="MuiInputBase-input MuiInput-input"
                id="realtimeTurnTime"
                mask="11 11 11 11"
                maxLength={11}
                name="realtimeTurnTime"
                onChange={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onPaste={[Function]}
                placeholder="DD HH MM SS"
                size={11}
                value="00 00 10 00"
              />
            </div>
          </div>
          <div
            className="group"
          >
            <label
              className="label"
              htmlFor="realtimeTurnTime"
            >
              Time warning at (HH MM SS) remaining
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline"
            >
              <input
                className="MuiInputBase-input MuiInput-input"
                id="timeWarning"
                mask="11 11 11"
                maxLength={8}
                name="timeWarning"
                onChange={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onPaste={[Function]}
                placeholder="HH MM SS"
                size={8}
                value="00 01 00"
              />
            </div>
          </div>
          <div
            className="hidden"
          >
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline"
              onClick={[Function]}
            >
              <input
                className="MuiInputBase-input MuiInput-input"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="text"
              />
            </div>
          </div>
          <div>
            <label
              className="MuiFormControlLabel-root"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-2 Mui-checked MuiIconButton-colorPrimary"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <input
                    checked={true}
                    className="PrivateSwitchBase-input-4"
                    data-indeterminate={false}
                    onChange={[Function]}
                    type="checkbox"
                    value="1"
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </span>
              </span>
              <span
                className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                Show Access codes
              </span>
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
